app/Http/Controllers/JobController.php

```diff
@@ -168,6 +166,121 @@ function ($value, $key) {
+            'url' => url('/'),
+            'inLanguage' => app()->getLocale(),
+            'applicantLocationRequirements' => [
+                '@type' => 'Country',
+                'sameAs' => 'https://www.wikidata.org/wiki/Q16',
+                'name' => 'Canada'
+            ],
+            'applicationContact' => [
+                '@type' => 'ContactPoint',
+                'email' => 'talent.cloud-nuage.de.talents@tbs-sct.gc.ca'
+            ],
+            'baseSalary' => [
+                '@type' => 'MonetaryAmount',
+                'currency' => 'CAD',
+                'minValue' => $jobPoster['salary_min'],
+                'maxValue' => $jobPoster['salary_max'],
+            ],
+            'datePosted' => $jobPoster['open_date_time'],
+            'employerOverview' => $jobLang['structured_data']['employer_overview'],
+            'employmentType' => $jobLang['structured_data']['employment_type'],
+            'estimatedSalary' => [
+                '@type' => 'MonetaryAmount',
+                'currency' => 'CAD',
+                'minValue' => $jobPoster['salary_min'],
+                'maxValue' => $jobPoster['salary_max']
+            ],
+            'hiringOrganization' => [
+                '@type' => 'GovernmentOrganization',
+                'name' => $jobPoster['manager']['user']['department']['name'],
+                'parentOrganization' => [
+                    '@type' => 'GovernmentOrganization',
+                    'name' => $jobLang['structured_data']['parent_organization'],
+                    'url' => $gocLang['logo_link'],
+                    'logo' => asset('/images/logo_canada_colour.png'),
+                ]
+            ],
+            'industry' => $jobLang['structured_data']['industry'],
+            'jobBenefits' => $benefitsString,
+            'jobLocation' => [
+                '@type' => 'Place',
+                'address' => [
+                    '@type' => 'PostalAddress',
+                    'addressLocality' => $jobPoster['city'],
+                    'addressRegion' => $jobPoster['province']['value'],
+                    'addressCountry' => 'CA'
+                ]
+            ],
+            'jobStartDate' => $jobPoster['start_date_time'],
+            'qualifications' => $jobPoster['education'],
+            'responsibilities' => $tasksString,
+            'salaryCurrency' => 'CAD',
+            'securityClearanceRequirement' => $securityClearanceString,
+            'skills' => $skillsString,
+            'specialCommitments' => $jobLang['structured_data']['special_commitments'],
+            'title' => $jobPoster['title'],
+            'totalJobOpenings' => '1',
+            'validThrough' => $jobPoster['close_date_time'],
+            'description' => $jobPoster['hire_impact'] ? nl2br($jobPoster['hire_impact']) : '',
+        ];
+
+        if ($jobPoster['remote_work_allowed'] == true) {
+            $structuredData['jobLocationType'] = 'TELECOMMUTE';
+        };
```

Comment:
I think I'd rather see this in a protected method within the controller since we already have a lot of logic going on within this particular action

Reply:
Please see [this commit](https://github.com/GCTC-NTGC/TalentCloud/pull/4537/commits/dcdb64df562be617a7026774fd735af0998ce9c7) where I have moved this into a protected method within the JobController class.